modules/RTC/RTC.js

```diff
@@ -49,6 +49,83 @@ function createLocalTracks(tracksInfo, options) {
 }
 
 /**
+ * Creates a {@code JitsiLocalTrack} instance.
+ *
+ * @param {Object} trackInfo
+ * @param {MediaStream} trackInfo.stream - The media stream instance that holds
+ * a track with audio or video.
+ * @param {MediaTrack} trackInfo.track - The actual MediaTrack instance that has
+ * audio or video that is in the passed in stream.
+ * @param {string} trackInfo.videoType - Whether the video track is "camera" or
+ * "desktop". Defaults to null.
+ * @param {string} trackInfo.sourceId - The id of the desktopsharing source.
+ * @param {string} trackInfo.sourceType - From which screensharing type the
+ * desktopsharing source is coming from.
+ * @param {Object} options - Additional information about the track. This
+ * parameter is left in from the previous createLocalTracks function
+ * specifically for electron, which may run an older version of chrome that does
+ * not have support for MediaStreamTrack#getSettings.
+ * @param {string} options.cameraDeviceId - The device id of the camera used for
+ * video capture. Used as a fallback if the information is not obtainable from
+ * the track.
+ * @param {string} options.facingMode - For mobile, which camera is being used.
+ * Used as a fallback if the information is not obtainable from the track.
+ * @param {string} options.micDeviceId - The device id of the microphone used
+ * for audio capture. Used as a fallback if the information is not obtainable
+ * from the track.
+ */
+function _newCreateLocalTrack({
+    sourceId,
+    sourceType,
+    stream,
+    track,
+    videoType = null
+}, options) {
+    let settingsFromTrack;
+
+    try {
```

Comment:
no try/catch

Reply:
I just tried chrome 58 on electron by making it be detected as chrome and it does not have getSettings. Whenever older chrome (or electron) gets put into the new gum flow it'll need truthy checks for getSettings. The try/catches were remnants from wip when we didn't take a browser-specific approach with adapter and needed to get around native throwing an error.